From: Ryan Lee <ryan.lee@canonical.com>
Date: Thu, 19 Feb 2026 14:53:23 -0800
Subject: [PATCH] libapparmor: move aa_get_lsm_iface decl in libapparmor.i to
 correct place

This function returns a string or None upon receiving an int, instead of
using errno conventions. Thus, it should be placed above the %exception
declaration used for functions invoking errno conventions.

Signed-off-by: Ryan Lee <ryan.lee@canonical.com>
---
 libraries/libapparmor/swig/SWIG/libapparmor.i | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/libapparmor/swig/SWIG/libapparmor.i b/libraries/libapparmor/swig/SWIG/libapparmor.i
index 49154fb..d14166d 100644
--- a/libraries/libapparmor/swig/SWIG/libapparmor.i
+++ b/libraries/libapparmor/swig/SWIG/libapparmor.i
@@ -170,6 +170,7 @@ warnings.warn("free_record is now a no-op as the record's memory is handled auto
   }
 }
 
+extern const char *aa_get_lsm_iface_name(int op);
 extern char *aa_splitcon(char *con, char **mode);
 
 #ifdef SWIGPYTHON
@@ -378,7 +379,6 @@ extern int aa_stack_onexec(const char *profile);
 extern int aa_find_mountpoint(char **mnt);
 extern int aa_getprocattr(pid_t tid, const char *attr, char **label, char **mode);
 
-extern const char *aa_get_lsm_iface_name(int op);
 extern int aa_get_self_attr(int op_type, char** label, char** mode);
 extern int aa_set_self_attr(int op_type, char *buf, int len);
 
